Error building datatable on macOS 10.15.5

Tried installing datatable from github using `pip install git+https://github.com/h2oai/datatable` per [the docs](https://datatable.readthedocs.io/en/latest/install.html#build-from-source), but I received the following error

```
Error linking `src/datatable/lib/_datatable.cpython-37m-darwin.so`

ERROR: Command errored out with exit status 1: /Users/bgorman/anaconda3/envs/datasci/bin/python /Users/bgorman/anaconda3/envs/datasci/lib/python3.7/site-packages/pip/_vendor/pep517/_in_process.py prepare_metadata_for_build_wheel /var/folders/vq/km5xms9179s_6vhpw5jxfrth0000gn/T/tmpqimvr62o Check the logs for full command output.
```

Any advice on how to get past this error? I'm using maxOS 10.15.5 with python 3.7.7 running on anaconda.

Thanks for the advice. Unfortunately, installing with `-v` doesn't highlight any glaring issues, and I get the same linking error when attempting to install `make wheel` from a local clone of the repo.

I tried installing with gcc instead of clang, via
```
export CC=gcc-9 
export CXX=g++-9
pip install git+https://github.com/h2oai/datatable
```

Interestingly, this returned a different error,
```
 In file included from src/core/python/obj.h:28,
    |                  from src/core/python/bool.h:24,
    |                  from src/core/python/_all.h:24,
    |                  from src/core/parallel/thread_pool.cc:22:
    | src/core/types.h: In member function 'bool CString::operator==(const CString&) const':
    | src/core/types.h:54:19: error: 'strncmp' is not a member of 'std'; did you mean 'dt::strncmp'?
    |    54 |             (std::strncmp(ch, other.ch, static_cast<size_t>(size)) == 0));
```

I guess I can use v 0.10.1 which installs fine on my machine. Unfortunately, it seems y'all have made a ton of progress since this version, so I'll look forward to the next release on PyPi!

--- a/src/core/expr/fbinary/pyfn.cc
+++ b/src/core/expr/fbinary/pyfn.cc
@@ -83,7 +83,7 @@ void py::DatatableModule::init_fbinary()
 {
   #define FBINARY(ARGS, OP) \
     ADD_FN(&dt::expr::fbinary_pyfn, dt::expr::ARGS); \
-    register_args(dt::expr::ARGS, dt::expr::OP);
+    register_args(dt::expr::ARGS, dt::expr::OP)
 
   FBINARY(args_atan2,      Op::ARCTAN2);
   FBINARY(args_hypot,      Op::HYPOT);

--- a/src/core/expr/fnary/pyfn.cc
+++ b/src/core/expr/fnary/pyfn.cc
@@ -126,7 +126,7 @@ void py::DatatableModule::init_fnary()
 {
   #define FNARY(ARGS, OP) \
     ADD_FN(&fnary_pyfn, dt::expr::ARGS); \
-    register_args(dt::expr::ARGS, dt::expr::OP);
+    register_args(dt::expr::ARGS, dt::expr::OP)
 
   FNARY(args_rowall,   Op::ROWALL);
   FNARY(args_rowany,   Op::ROWANY);

--- a/src/core/expr/funary/pyfn.cc
+++ b/src/core/expr/funary/pyfn.cc
@@ -113,7 +113,7 @@ void py::DatatableModule::init_funary()
 {
   #define FUNARY(ARGS, OP) \
     ADD_FN(&dt::expr::funary_pyfn, dt::expr::ARGS); \
-    register_args(dt::expr::ARGS, dt::expr::OP);
+    register_args(dt::expr::ARGS, dt::expr::OP)
 
   // Basic
   FUNARY(args_len,     Op::LEN);

--- a/src/core/models/murmurhash.cc
+++ b/src/core/models/murmurhash.cc
@@ -45,7 +45,7 @@ uint64_t hash_murmur2(const void* key, uint64_t len)
     case 2: h ^= uint64_t(data2[1]) << 8;  FALLTHROUGH;
     case 1: h ^= uint64_t(data2[0]);
             h *= m;
-  };
+  }
 
   h ^= h >> r;
   h *= m;
@@ -113,7 +113,7 @@ void hash_murmur3(const void* key, const uint64_t len, void* out) {
     case  2: k1 ^= (static_cast<uint64_t>(tail[ 1])) << 8; FALLTHROUGH;
     case  1: k1 ^= (static_cast<uint64_t>(tail[ 0])) << 0;
              k1 *= c1; k1 = ROTL64(k1, 31); k1 *= c2; h1 ^= k1;
-  };
+  }
 
   //----------
   // finalization

--- a/src/core/python/ext_module.h
+++ b/src/core/python/ext_module.h
@@ -84,7 +84,7 @@ class ExtModule {
   add(                                                                         \
     [](PyObject*, PyObject* args, PyObject* kwds) -> PyObject* {               \
       return ARGS.exec_function(args, kwds, FUNCTION);                         \
-    }, ARGS);                                                                  \
+    }, ARGS)
 
 
 } // namespace py

--- a/src/core/stype.h
+++ b/src/core/stype.h
@@ -240,16 +240,16 @@ using read_t = typename _rdt<s>::t;
   * the "most typical" SType that represents type T.
   */
 template <typename T> static constexpr SType _sfr = SType::VOID;
-template <> constexpr SType _sfr<bool>      = SType::BOOL;
-template <> constexpr SType _sfr<int8_t>    = SType::INT8;
-template <> constexpr SType _sfr<int16_t>   = SType::INT16;
-template <> constexpr SType _sfr<int32_t>   = SType::INT32;
-template <> constexpr SType _sfr<int64_t>   = SType::INT64;
-template <> constexpr SType _sfr<float>     = SType::FLOAT32;
-template <> constexpr SType _sfr<double>    = SType::FLOAT64;
-template <> constexpr SType _sfr<CString>   = SType::STR32;
-template <> constexpr SType _sfr<PyObject*> = SType::OBJ;
-template <> constexpr SType _sfr<py::robj>  = SType::OBJ;
+template <> static constexpr SType _sfr<bool>      = SType::BOOL;
+template <> static constexpr SType _sfr<int8_t>    = SType::INT8;
+template <> static constexpr SType _sfr<int16_t>   = SType::INT16;
+template <> static constexpr SType _sfr<int32_t>   = SType::INT32;
+template <> static constexpr SType _sfr<int64_t>   = SType::INT64;
+template <> static constexpr SType _sfr<float>     = SType::FLOAT32;
+template <> static constexpr SType _sfr<double>    = SType::FLOAT64;
+template <> static constexpr SType _sfr<CString>   = SType::STR32;
+template <> static constexpr SType _sfr<PyObject*> = SType::OBJ;
+template <> static constexpr SType _sfr<py::robj>  = SType::OBJ;
 
 template <typename T>
 static constexpr SType stype_from = _sfr<T>;

--- a/src/core/stype.h
+++ b/src/core/stype.h
@@ -239,20 +239,21 @@ using read_t = typename _rdt<s>::t;
   * Approximate inverse of `read_t<SType>`: given a type T, returns
   * the "most typical" SType that represents type T.
   */
-template <typename T> static constexpr SType _sfr = SType::VOID;
-template <> static constexpr SType _sfr<bool>      = SType::BOOL;
-template <> static constexpr SType _sfr<int8_t>    = SType::INT8;
-template <> static constexpr SType _sfr<int16_t>   = SType::INT16;
-template <> static constexpr SType _sfr<int32_t>   = SType::INT32;
-template <> static constexpr SType _sfr<int64_t>   = SType::INT64;
-template <> static constexpr SType _sfr<float>     = SType::FLOAT32;
-template <> static constexpr SType _sfr<double>    = SType::FLOAT64;
-template <> static constexpr SType _sfr<CString>   = SType::STR32;
-template <> static constexpr SType _sfr<PyObject*> = SType::OBJ;
-template <> static constexpr SType _sfr<py::robj>  = SType::OBJ;
+template <typename T>
+            inline constexpr SType _sfr()            { return SType::VOID; }
+template <> inline constexpr SType _sfr<bool>()      { return SType::BOOL; }
+template <> inline constexpr SType _sfr<int8_t>()    { return SType::INT8; }
+template <> inline constexpr SType _sfr<int16_t>()   { return SType::INT16; }
+template <> inline constexpr SType _sfr<int32_t>()   { return SType::INT32; }
+template <> inline constexpr SType _sfr<int64_t>()   { return SType::INT64; }
+template <> inline constexpr SType _sfr<float>()     { return SType::FLOAT32; }
+template <> inline constexpr SType _sfr<double>()    { return SType::FLOAT64; }
+template <> inline constexpr SType _sfr<CString>()   { return SType::STR32; }
+template <> inline constexpr SType _sfr<PyObject*>() { return SType::OBJ; }
+template <> inline constexpr SType _sfr<py::robj>()  { return SType::OBJ; }
 
 template <typename T>
-static constexpr SType stype_from = _sfr<T>;
+static constexpr SType stype_from = _sfr<T>();
 
 
 /**

